packages/qwik/src/core/platform/platform.ts

```diff
@@ -65,9 +65,10 @@ export const createPlatform = (): CorePlatform => {
  * @returns Fully qualified URL.
  */
 export const toUrl = (doc: Document, containerEl: QwikElement, url: string | URL): URL => {
-  const baseURI = doc.baseURI;
+  const baseURI = new URL(doc.baseURI);
   const base = new URL(containerEl.getAttribute('q:base') ?? baseURI, baseURI);
-  return new URL(url, base);
+  const pathUrl = (base.pathname + url).replace(/\/+/g, '/');
```

Comment:
`url` can be anything, including query string and hash, so you shouldn't run replace on it.
For example `toUrl(doc, el, "/hello/foo?stuff=///meep#123")` should work.
Just check if url isn't empty and starts with /